refract-spec.md

```diff
@@ -137,9 +139,481 @@ This is how it would expressed in the compact version.
+- (object)
+    - `prefix` (string) - A prefix
+    - `href` (string) - An `href` is a resolvable URL to a namespace
+
+#### Example Using Both String and Object Namespace Elements
+
+This example shows a namespace being included by way of a string, and a prefixed namespace.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "namespaces": [
+      "http://example.com/namespace1",
+      { "prefix": "ns2", "href": "http://example.com/namespace2" }
+    ]
+  }
+}
+```
+
+### Prefix (Element)
+
+A prefix is a way to reference elements in prefixed namespaces.
+
+#### Example
+
+This example says the `foo` element is found in the prefixed namespaced prefixed as `ns2`.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "prefix": "ns2"
+  },
+  "content": "baz"
+}
+```
+
+## Extending Elements
+
+Elements by default do not include any of the attributes or content from the element in which they reference. An element MAY be extended in order to inherit from other elements.
+
+### Extend (Element)
+
+The `extend` element provides a way to a way to multiply inherit one or more elements to form a new element. The `extend` element MUST NOT affect the elements being extended, but MUST define a new instance of an element.
+
+The `extend` element MUST do a deep merge of the elements found in the `content`, but MUST NOT include any IDs in the final element. Each element found in the `content` MUST be of the same element type.
+
+The elements are merged from left to right, meaning the latter elements are merged into the former elements.
+
+#### Examples
+
+```json
+{
+  "element": "extend",
+  "content": [
+    {
+      "element": "foo",
+      "attributes": {
+        "baz": "bar"
+      },
+      "content": "first"
+    },
+    {
+      "element": "foo",
+      "content": "second"
+    }
+  ]
+}
+```
+
+The value for this new element would be the following.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "baz": "bar"
+  },
+  "content": "second"
+}
+```
+
+Elements MAY also be referenced when using extend. Here is an element defined and given an ID of `bar`.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "id": "bar"
+  },
+  "content": "second"
+}
+```
+
+Here is an extend element referencing the previously `bar` element.
+
+```json
+{
+  "element": "extend",
+  "content": [
+    {
+      "element": "foo",
+      "content": "first"
+    },
+    {
+      "element": "ref",
+      "content": "#bar"
+    }
+  ]
+}
+```
+
+The resulting element would be the following. Note that the `id` was not included in the final element.
+
+```json
+{
+  "element": "foo",
+  "content": "second"
+}
+```
+
+## Creating Element Types
+
+Element types MAY be referenced or extended to create new element types. If given an ID, these elements MAY be used within the local namespace or referenced from other namespaces if accessible from a resolvable URL. Element types defined within a namespace take priority over all other namespaced types.
```

Comment:
I think you need to re-enforce how you reference these custom types and the rules for finding them locally or in namespaces. May be another section in the spec that is better.